src/utils/narrow.js

```diff
@@ -399,14 +399,28 @@ export const isSearchNarrow = (narrow?: Narrow): boolean =>
 /**
  * Convert the narrow into the form used in the Zulip API at get-messages.
  */
-export const apiNarrowOfNarrow = (narrow: Narrow): ApiNarrow =>
+export const apiNarrowOfNarrow = (
+  narrow: Narrow,
+  allUsersById: Map<number, UserOrBot>,
+): ApiNarrow =>
   caseNarrow(narrow, {
     stream: streamName => [{ operator: 'stream', operand: streamName }],
     topic: (streamName, topic) => [
       { operator: 'stream', operand: streamName },
       { operator: 'topic', operand: topic },
     ],
-    pm: emails => [{ operator: 'pm-with', operand: emails.join(',') }],
+    pm: (emails_, ids) => {
+      const emails = [];
+      for (const id of ids) {
+        const email = allUsersById.get(id)?.email;
+        if (email === undefined) {
+          throw new Error('apiNarrowOfNarrow: missing user');
+        }
+        emails.push(email);
+      }
+      // TODO(server-2.1): just send IDs instead
```

Comment:
If it makes a big difference to the amount of work the server has to do, we could check the stored server version in Redux and courteously send IDs to the newer servers, while still sending emails to older ones. But it might not make much of a difference. 🙂

Reply:
Yeah, I don't think it materially affects the work the server has to do. It'll want to look up the users in the database in any event (for one thing, to make sure they're valid users that are in your realm); and the database will have indexes so that looking up by ID or by email are both fast, because those are both common operations.